File: python/burkut/resources/bonds.py
"""
Government & Corporate Bonds Resource
"""

from typing import Any, Dict, List
from burkut.resources.base import BaseResource


class BondsResource(BaseResource):
    """Tahvil ve Bono verileri."""

    def list(self) -> List[Dict[str, Any]]:
        """
        Tüm tahvil ve bono verilerini listeler.

        Returns:
            Tahvil verileri listesi.
        """
        res = self._get("/bonds")
        return res if isinstance(res, list) else []

    def get(self, symbol: str) -> Dict[str, Any]:
        """
        Belirli bir tahvilin detaylarını getirir.

        Args:
            symbol: Tahvil kodu
        """
        if not symbol or not isinstance(symbol, str):
            raise ValueError("symbol parametresi boş olamaz.")
        return self._get(f"/bonds/{symbol.strip().upper()}")

File: nodejs/src/resources/bonds.ts
import { BaseResource } from './base';
import type { BondItem } from '../types';

export class BondsResource extends BaseResource {
  /**
   * Tüm tahvil ve bono verilerini listeler.
   */
  async list(): Promise<BondItem[]> {
    return this._get<BondItem[]>('/bonds');
  }

  /**
   * Belirli bir tahvilin detaylarını getirir.
   * @param symbol Tahvil kodu
   */
  async get(symbol: string): Promise<BondItem> {
    if (!symbol || typeof symbol !== 'string') {
      throw new Error('symbol parametresi geçerli bir metin olmalıdır.');
    }
    return this._get<BondItem>(`/bonds/${symbol.trim().toUpperCase()}`);
  }
}
